### PR TITLE
What's Here section for Integer

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: numeric.c [hunk 1]
```diff
@@ -3418,10 +3418,94 @@ rb_num2ull(VALUE val)
...
 /********************************************************************
  *
- * Document-class: Integer
+ *  Document-class: Integer
  *
- *  Holds Integer values.  You cannot add a singleton method to an
- *  Integer object, any attempt to do so will raise a TypeError.
+ *  An \Integer object represents an integer value.
+ *
+ *  An attempt to add a singleton method to this class
```
Comment (jeremyevans) [inline numeric.c:3425]: `to this class` -> `to an instance of this class`

File: numeric.c [hunk 2]
```diff
@@ -3418,10 +3418,94 @@ rb_num2ull(VALUE val)
...
+ * - #allbits?:: Returns whether all bits in +self+ are set.
+ * - #anybits?:: Returns whether any bits in +self+ are set.
+ * - #nobits?:: Returns whether no bits in +self+ are set.
+ *
+ * === Comparing
+ *
+ * - {<}[#method-i-3C]:: Returns whether +self+ is less than the given value.
+ * - {<=}[#method-i-3C-3D]:: Returns whether +self+ is less than
+ *                           or equal to the given value.
+ * - {<=>}[#method-i-3C-3D-3E]:: Returns:
```
Comment (jeremyevans) [inline numeric.c:3452]: This is probably too long.  Maybe `Returns a number indicating whether the +self+ is less than, equal to, or greater than the given value`.

File: numeric.c [hunk 3]
```diff
@@ -3418,10 +3418,94 @@ rb_num2ull(VALUE val)
...
+ * - {>}[#method-i-3E]:: Returns whether +self+ is greater than the given value.
+ * - {>=}[#method-i-3E-3D]:: Returns whether +self+ is greater than
+ *                           or equal to the given value.
+ *
+ * === Converting
+ *
+ * - ::sqrt:: Returns the integer square root of the given value.
+ * - ::try_convert:: Returns the given value converted to an \Integer.
+ * - #% (aliased as #modulo):: Returns +self+ modulo the given value.
+ * - {&}[#method-i-26]:: Returns the bitwise logical AND of +self+ and the given value.
```
Comment (jeremyevans) [inline numeric.c:3470]: `bitwise logical` -> `bitwise`.  Same for other entries in this section.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good.  See inline comments.  This can be committed without further review if each comment is addressed.

File: numeric.c (hunk 1 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

File: numeric.c (hunk 3 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

File: numeric.c (hunk 2 above)
Comment (BurdetteLamar) [reply numeric.c]: Done.

**Pull request merged**